### ISSUE TITLE
Element and attribute were switched for abbr

### ISSUE THREAD

**Issue by JAWS-test** [w3c/html-aam#246]
The description for the <code>abbr</code> element applies to the <code>abbr</code> attribute and not to the element:

>  "abbr" attribute on the containing td if a single child, text content used as a value 

https://w3c.github.io/html-aam/#html-element-role-mappings

**Comment by scottaohara** [comment]
yup, this needs fixing.  apparently has been like that for quite some time now.  will do.

**Comment by scottaohara** [comment]
@jcsteh in looking to clear this up, I was talking to @joanmarie about how ATK exposes `abbr` with a `title` attribute, presently IA2/MSAA implementations expose the `title` value as the name for the element.  Honestly I'm not sure this should be the case, since it seems that most AT don't respect this (by default at least), nor do I think we want an abbreviation to be suppressed by its long form name.  

We're very interested in how you think this should be exposed / mapped.

Thanks

edited for clarity, per @joanmarie's comment below.

**Comment by joanmarie** [comment]
Point of clarification: The *implementations* expose it that way for ATK and IA2/MSAA. There's nothing (that I am aware of) in these APIs that dictate this exposure.

**Comment by jcsteh** [comment]
Hmm. This is tricky. (I feel like i say that a lot here.)

TL;DR: I think we should map abbr title to acc description, but we should check with AT vendors; this is a backwards incompatible change.

Regarding AT support, for NVDA at least, you can query the object beneath the cursor, which means you can hear the expanded abbreviation (title) if you wish. The presence of an abbr with title isn't discoverable, though. (I'd argue it isn't discoverable visually either - you have to mouse over it to know it's there - but that's probably getting beyond the scope of this issue.)

That aside, I agree that mapping this to name seems weird. I tend to think of name as primary content and description as secondary. The current mapping doesn't fit that definition; the abbreviated text (content) should be primary and the expanded abbreviation (title) should be secondary.

What makes this tricky is that right now, abbr doesn't calculate its name from subtree, consistent with `<p>`, etc., and I'd argue this is correct behaviour. The problem is that this means we'll end up with the abbr accessible having no name but a description. That seems weird and is inconsistent with how we handle title everywhere else (expose as acc name if no other name, else expose as acc description).

Still, I guess the spec does explicitly call out the title attribute for abbr, so this "special treatment" is probably okay.

**Comment by scottaohara** [comment]
to consider with this issue - https://github.com/w3c/aria/issues/816
